Tasks/AzureRmWebAppDeployment/AzureRmWebAppDeployment.ts

```diff
@@ -0,0 +1,235 @@
+        var silentCommand = '@echo off \n';
+        var msDeployCommand = '"' + msDeployPath + '" ' + msDeployCmdArgs;
+        var batchCommand = silentCommand + msDeployCommand;
+
+        tl.writeFile(msDeployBatchFile, batchCommand);
+        tl._writeLine(tl.loc("Runningcommand", msDeployCommand));
+        await tl.exec("cmd", ['/C', msDeployBatchFile], <any> {failOnStdErr: true});
+        tl._writeLine(tl.loc('WebappsuccessfullypublishedatUrl0', publishingProfile.destinationAppUrl));
+    }
+    catch(error) {
+        tl.error(tl.loc('Failedtodeploywebsite'));
+        isDeploymentSuccess = false;
+        deploymentError = error;
+    }
+
+    try {
+        tl._writeLine(await azureRmUtil.updateDeploymentStatus(publishingProfile, isDeploymentSuccess));
+    }
+    catch(error) {
+        tl.warning(error);
+    }
+    finally {
+        if(!isDeploymentSuccess) {
+            throw Error(deploymentError);
+        }
+    }
+}
+
+/**
+ * Deploys website using Kudu REST API
+ * 
+ * @param   webDeployPkg                   Web deploy package
+ * @param   webAppName                     Web App Name
+ * @param   publishingProfile              Azure RM Connection Details
+ * @param   virtualApplication             Virtual Application Name
+ * @param   isFolderBasedDeployment        Input is folder or not
+ *
+ */
+async function DeployUsingKuduDeploy(webDeployPkg, azureWebAppDetails, publishingProfile, virtualApplication, isFolderBasedDeployment) {
+
+    var isDeploymentSuccess = true;
+    var deploymentError = null;
+
+    try {
+        var virtualApplicationMappings = azureWebAppDetails.properties.virtualApplications;
+        var webAppZipFile = webDeployPkg;
+        if(isFolderBasedDeployment) {
+            webAppZipFile = await kuduUtility.archiveFolder(webDeployPkg);
+            tl.debug(tl.loc("Compressedfolderintozip", webDeployPkg, webAppZipFile));
+        } else {
+            if (await kuduUtility.containsParamFile(webAppZipFile)) {
+                throw new Error(tl.loc("MSDeploygeneratedpackageareonlysupportedforWindowsplatform")); 
+            }
+        }
+        var pathMappings = kuduUtility.getVirtualAndPhysicalPaths(virtualApplication, virtualApplicationMappings);
+        await kuduUtility.deployWebAppPackage(webAppZipFile, publishingProfile, pathMappings[0], pathMappings[1]);
+        tl._writeLine(tl.loc('WebappsuccessfullypublishedatUrl0', publishingProfile.destinationAppUrl));
+    }
+    catch(error) {
+        tl.error(tl.loc('Failedtodeploywebsite'));
+        isDeploymentSuccess = false;
+        deploymentError = error;
+    }
+
+    try {
+        tl._writeLine(await azureRmUtil.updateDeploymentStatus(publishingProfile, isDeploymentSuccess));
+    }
+    catch(error) {
+        tl.warning(error);
+    }
+    finally {
+        if(!isDeploymentSuccess) {
+            throw Error(deploymentError);
```

Comment:
Move this code out of finally block as this not at all related to previous try/catch